cpp/dolfinx/fem/Form.h

```diff
@@ -277,6 +292,43 @@ class Form
       throw std::runtime_error("No mesh entities for requested domain index.");
   }
 
+  /// @brief Compute the list of entity indices for the ith integral (kernel)
+  /// of a given type. The indices are mapped from the integration domain mesh
+  /// to the mesh `mesh`.
+  ///
+  /// @param type Integral type
+  /// @param i Integral ID, i.e. (sub)domain index
+  /// @param mesh The mesh to map the entities to
+  /// @return List of active entities in `mesh` for the given integral
+  std::vector<std::int32_t>
+  domain(IntegralType type, int i,
+         std::shared_ptr<const mesh::Mesh<U>> mesh) const
```

Comment:
This should be a `const mesh::Mesh<U>&`. Only pass `shared_ptr` to a class method that holds onto the pointer.